src/overlays/actors/ovl_En_Rat/z_en_rat.c

```diff
@@ -28,123 +44,910 @@ const ActorInit En_Rat_InitVars = {
+        this->collider.base.ocFlags1 &= ~OC1_HIT;
+        EnRat_Explode(this, play);
+        return;
+    }
+
+    this->actionFunc(this, play);
+
+    if ((this->actionFunc != EnRat_PostDetonation) && (this->actionFunc != EnRat_Revive)) {
+        if (this->stunTimer > 0) {
+            this->stunTimer--;
+        } else if (this->stunTimer < 0) {
+            if (this->stunTimer == -2) {
+                this->stunTimer = -1;
+            } else if ((this->actor.flags & ACTOR_FLAG_2000) != ACTOR_FLAG_2000) {
+                EnRat_Explode(this, play);
+                return;
+            }
+        } else if (this->actionFunc != EnRat_Bounced) {
+            if (this->actor.floorBgId != BGCHECK_SCENE) {
+                EnRat_HandleNonSceneCollision(this, play);
+            }
+
+            if (!this->hasLostTrackOfPlayer) {
+                EnRat_ChooseDirection(this);
+            }
+
+            if ((this->actionFunc != EnRat_SpottedPlayer) && !EnRat_IsTouchingFloor(this, play)) {
+                EnRat_Explode(this, play);
+                return;
+            }
+
+            if (this->shouldRotateOntoSurfaces) {
+                EnRat_UpdateRotation(this);
+                this->actor.shape.rot.x = -this->actor.world.rot.x;
+                this->actor.shape.rot.y = this->actor.world.rot.y;
+                this->actor.shape.rot.z = this->actor.world.rot.z;
+            }
+
+            Actor_MoveWithoutGravity(&this->actor);
+            this->actor.floorHeight = this->actor.world.pos.y;
+        } else {
+            Actor_MoveWithGravity(&this->actor);
+            Actor_UpdateBgCheckInfo(play, &this->actor, 20.0f, 30.0f, 60.0f, 7);
+        }
+
+        if (SurfaceType_IsWallDamage(&play->colCtx, this->actor.floorPoly, this->actor.floorBgId)) {
+            EnRat_Explode(this, play);
+            return;
+        }
+
+        this->collider.dim.worldSphere.center.x = this->actor.world.pos.x + (this->axisUp.x * 10.0f);
+        this->collider.dim.worldSphere.center.y = this->actor.world.pos.y + (this->axisUp.y * 10.0f);
+        this->collider.dim.worldSphere.center.z = this->actor.world.pos.z + (this->axisUp.z * 10.0f);
+
+        if (this->actionFunc != EnRat_Revive) {
+            if (this->stunTimer == 0) {
+                CollisionCheck_SetAT(play, &play->colChkCtx, &this->collider.base);
+            }
+
+            CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+            CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+        }
+
+        Actor_SetFocus(&this->actor, this->actor.shape.yOffset * 0.015f);
+    }
+}
+
+s32 EnRat_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnRat* this = THIS;
+
+    if (limbIndex == REAL_BOMBCHU_LIMB_BODY) {
+        pos->y -= this->revivePosY;
+    }
+
+    if (limbIndex == REAL_BOMBCHU_LIMB_TAIL_END) {
+        *dList = NULL;
+    }
```

Comment:
Weird. It's clear from the next function they just want this limb to draw the fuse, but why would they nullify the DL here 100% of the time when they could just make that limb null then add their thing next function? Does the skeleton in z64Utils actually show something on that tail limb?

Reply:
Yeah, it's a triangle:
![image](https://user-images.githubusercontent.com/12245827/179897862-37cfea23-808f-46f8-8a5a-abac438f8e06.png)

Your guess is as good as mine as to why they actually put something here.

Reply:
I can think of at least two possibilities:

They originally wanted a static bomb but it looked bad so they decided to make it dynamic after the fact.

Or maybe they wanted the bomb/fuse to be removable/disarmable with some item/form move, and they were going to give him some frizy tail left over after you remove the bomb, but they ran out of time and had to work on other things.

Well, a blue triangle isnt as cool but at least there is something there after all to nullify.